### PR TITLE
Use special "detached" index for manifest plugins

### PR DESCRIPTION
**Pull request by chriskim06**
Let me know if you want me to move this out to a package level variable, figured it would be ok like this since this situation is only encountered in this file.

Related issue: #483

### CODE REVIEW

**Comment by ahmetb** [comment]
> Let me know if you want me to move this out to a package level variable, figured it would be ok like this since this situation is only encountered in this file.

I think it’s fine.

**Comment by ahmetb** [comment]
Since we introduce a behavior, we should test.

Maybe add a test to integ tests of krewinstallmanifest ones loading receipt and checking source==detached.

**Comment by ahmetb** [comment]
> So let's extract it to `constants.DetachedIndexName` or similar.

But technically we’re offering these libraries as part of our API. Would anyone build tooling/automation around this const? If not, better not expose?

**Comment by chriskim06** [comment]
The integration test failed for some reason, I think I vaguely remember seeing this in other PRs. I'll investigate further later tonight.

**Comment by ahmetb** [comment]
> The integration test failed for some reason (worked locally), I think I vaguely remember seeing this in other PRs. I'll investigate further later tonight.

Seems like it was a flake (somehow), re-running the workflow worked. It happened at least once before. I'm suspecting GitHub's endpoint sending us an error response somehow. Need to look at download code to see if it's checking for non-200 codes.

File: integration_test/install_test.go [hunk 1]
```diff
@@ -216,3 +220,14 @@ func localTestServer() (string, func()) {
 	s := httptest.NewServer(http.FileServer(http.Dir("testdata")))
 	return s.URL, s.Close
 }
+
+func assertManifestPluginIndex(t *testing.T, test *ITest, plugin string) {
```
Comment (ahmetb) [inline integration_test/install_test.go]: assertPluginFromIndex is a better name potentially?
also why is this not defined on `test` (like assertExecutableInPath)? that way you don't have to pass it `t`.

File: integration_test/install_test.go [hunk 2]
```diff
@@ -216,3 +220,14 @@ func localTestServer() (string, func()) {
...
+	receiptPath := test.TempDir().Path("receipts/" + plugin + constants.ManifestExtension)
```
Comment (ahmetb) [inline integration_test/install_test.go]: don't guess filesystem layout; use environment.Paths (unless you're actually testing user-visible behavior, breaking behavior between versions, or Paths type).

File: integration_test/install_test.go (hunk 1 above)
Comment (chriskim06) [reply integration_test/install_test.go]: I can add it there. I added this to just check "detached" and thought that would be a little too specific to `install --manifest` to add to `testutil_test.go`.

**Comment by chriskim06** [comment]
@corneliusweig @ahmetb let me know what you guys think about "detached". There's now a few usages of it in the integration test so would a good compromise be to make it a public const in the cmd package so that its not exposed but still available within the code? Although I think that might be a little strange seeing `cmd.DetachedIndexName`.

**Comment by ahmetb** [comment]
@chriskim06 nothing wrong with actually duplicating those in integration_test.

In fact, tests should rely on consts less and less. Because we want to test "behavior". If someone changes the value of the const, that should break the tests to notify us of behavior change, so please refrain from using the const in the tests.

**Comment by ahmetb** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**